File: 2022-2023/sourcecode/lab/lab4/DynamicFragmentJava/app/src/main/java/pl/edu/uwr/pum/dynamicfragmentjava/MainFragment.java
package pl.edu.uwr.pum.dynamicfragmentjava;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import java.util.Objects;

public class MainFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_main, container, false);
        view.findViewById(R.id.start_fragment_a)
                .setOnClickListener(v -> requireActivity().getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.fragment_container_view_tag, new FragmentA())
                    .addToBackStack(null)
                    .commit());
        return view;
    }
}

File: 2022-2023/sourcecode/lab/lab4/DynamicFragmentKotlin/app/src/main/java/pl/edu/uwr/pum/dynamicfragmentkotlin/MainFragment.kt
package pl.edu.uwr.pum.dynamicfragmentkotlin

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment

class MainFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_main, container, false)
        view.findViewById<Button>(R.id.start_fragment_a)
            .setOnClickListener { requireActivity().supportFragmentManager
                .beginTransaction()
                .replace(R.id.fragment_container_view_tag, FragmentA())
                .addToBackStack(null)
                .commit()
            }
        return view
    }
}